src/common/selectors/index.js

```diff
@@ -1,12 +1,65 @@
 import { createSelector } from 'reselect';
 
+import { parseGitHubRepoUrl } from '../helpers/github-url';
+
 const getSnaps = state => state.snaps;
+const getSnapBuilds = state => state.snapBuilds;
+
+// returns bool - true if there are any snaps in state
+export const hasSnaps = createSelector(
+  [getSnaps],
+  (snaps) => snaps.snaps ? snaps.snaps.length > 0 : false
+);
 
+// returns bool - true if there are no snaps in state
+export const hasNoSnaps = createSelector(
+  [getSnaps],
+  (snaps) => snaps.snaps ? snaps.snaps.length === 0 : true
+);
+
+// returns bool - true if there are no snaps with registered name in state
 export const hasNoRegisteredNames = createSelector(
   [getSnaps],
   (snaps) => {
-    return !snaps.snaps.some((snap) => {
+    return snaps.snaps ? !snaps.snaps.some((snap) => {
```

Comment:
This catches the case where snaps.snaps is null, as it is in initial state? Oops, naughty me. :(

I think it would be simpler if initial/default state of lists was [] rather than null, but that's for another day.